source/ui/pixel-display-painted-item.cpp

```diff
@@ -0,0 +1,738 @@
+                          std::size_t y) const {
+  image->setPixel(x, y, getPixel(memoryAccess, x, y));
+}
+void Options::updateColor(Optional<OutputComponent *> memoryAccess,
+                          std::shared_ptr<QImage> image,
+                          std::size_t index) const {
+  image->setColor(index, getColor(memoryAccess, index));
+}
+void Options::updateMemory(Optional<OutputComponent *> memoryAccess,
+                           std::shared_ptr<QImage> image,
+                           std::size_t address,
+                           std::size_t amount) const {
+  getColorMode().updateMemory(memoryAccess, *this, image, address, amount);
+}
+void Options::updateAllPixels(Optional<OutputComponent *> memoryAccess,
+                              std::shared_ptr<QImage> image) const {
+  getColorMode().updateAllPixels(memoryAccess, *this, image);
+}
+void Options::updateAllColors(Optional<OutputComponent *> memoryAccess,
+                              std::shared_ptr<QImage> image) const {
+  getColorMode().updateAllColors(memoryAccess, *this, image);
+}
+
+std::shared_ptr<QImage> Options::makeImage() const {
+  QImage::Format format = QImage::Format_RGB32;
+  switch (colorMode) {
+    case 0: format = QImage::Format_RGB32; break;
+    case 1: format = QImage::Format_Mono; break;
+    default: break;
+  }
+  return std::make_shared<QImage>(width, height, format);
+}
+
+std::size_t ColorMode::loadPointer(Optional<OutputComponent *> memoryAccess,
+                                   std::size_t address,
+                                   bool indirect,
+                                   std::size_t cellSize,
+                                   std::size_t pointerSize) {
+  if (indirect) {
+    if (memoryAccess) {
+      MemoryValue pointer = ColorMode::getMemoryValueAt(
+          memoryAccess, address, (pointerSize + cellSize - 1) / cellSize);
+      return conversions::convert<std::size_t>(
+          pointer, conversions::standardConversions::nonsigned);
+    } else {
+      return 0;
+    }
+  } else {
+    return address;
+  }
+}
+MemoryValue
+ColorMode::getMemoryValueAt(Optional<OutputComponent *> memoryAccess,
+                            std::size_t address,
+                            std::size_t length,
+                            std::size_t defaultLength) {
+  if (memoryAccess) {
+    if ((*memoryAccess)->getMemoryAccess().getMemorySize().get() >=
+        address + length) {
+      return (*memoryAccess)
+          ->getMemoryAccess()
+          .tryGetMemoryValueAt(address, length)
+          .get();
+    }
+  }
+  return MemoryValue{defaultLength};
+}
+
+
+// RGB*-_-*-_-*-_-*-_-*-_-*-_-*-_-*-_-*-_-*-_-*-_-*-_-*-_-*-_-*-_-*-_-*-_-*-_-*
+const ColorMode::GetPixelFunction ColorMode::RGBGetPixel = [](
+    Optional<OutputComponent *> memoryAccess,
+    const Options &o,
+    std::size_t x,
+    std::size_t y) -> std::uint32_t {
+  std::size_t cellSize = 8;    // TODO
+  std::size_t pointerSize = 32;// TODO
```

Comment:
cellsize probably should be got from memory, pointersize could possibly be a very advancedly advanced option